### PR TITLE
Merge latest upstream and update vendoring for cluster-api

### PR DESCRIPTION
**Pull request by stbenjam**
See: https://github.com/metal3-io/cluster-api-provider-baremetal/issues/88

I'm not sure if this should go to metal3-io first, the OpenStack PR went direct to the openshift fork.  The instructions for testing CAPB also don't seem to work for me, either.

I followed the info in docs/, but when running `./bin/machine-controller-manager` on my cluster it reports:

```
{"level":"info","ts":1560949145.6802013,"logger":"baremetal-controller-manager","msg":"Waiting for API group metal3.io/v1alpha1 to be available: server does not support API version \"metal3.io/v1alpha1\""}
```

### CODE REVIEW

**Comment by stbenjam** [comment]
@openshift-ci-robot laughed at me :disappointed:

File: cmd/manager/main.go [hunk 1]
```diff
@@ -59,7 +60,15 @@ func main() {
...
 	// Setup a Manager
-	mgr, err := manager.New(cfg, manager.Options{MetricsBindAddress: *metricsAddr})
+	opts := manager.Options{
+		MetricsBindAddress: *metricsAddr,
+	}
+	if *watchNamespace != "" {
+		opts.Namespace = *watchNamespace
+		klog.Infof("Watching machine-api objects only in namespace %q for reconciliation.", opts.Namespace)
+	}
+
```
Comment (russellb) [inline cmd/manager/main.go]: This doesn't seem OpenShift specific, so let's put it in the upstream version first

Comment (stbenjam) [reply cmd/manager/main.go]: https://github.com/metal3-io/cluster-api-provider-baremetal/pull/89

**Pull request merged**